pkg/rehearse/jobs.go

```diff
@@ -191,6 +218,26 @@ func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.
 	return rehearsals
 }
 
+func hasChangedTemplateVolume(volumeMounts []v1.VolumeMount, volumes []v1.Volume, templates config.CiTemplates) (bool, int, string) {
+	var templateKey string
+	var volumeName string
+
+	for _, volumeMount := range volumeMounts {
+		if _, ok := templates[volumeMount.SubPath]; ok {
+			templateKey = volumeMount.SubPath
+			volumeName = volumeMount.Name
+		}
+	}
+
+	for index, volume := range volumes {
+		if volume.Name == volumeName {
+			return true, index, templateKey
```

Comment:
So this looks for a first occurrence... Is it sufficient? Won't we need to fixup multiple references, maybe to a cluster profile **and** the template?

Reply:
We don't need to change the secrets. Only the template.

Reply:
OK